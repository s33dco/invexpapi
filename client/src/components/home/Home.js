/* eslint-disable react/require-default-props */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import About from './About';
import Dashboard from './Dashboard';

const Home = ({ isAuthenticated, user}) => {

	return (
		<div className="container">
			{isAuthenticated && user ? <Dashboard /> : <About />}
		</div>
	);
};

Home.propTypes = {
	isAuthenticated: PropTypes.bool.isRequired,
	user: PropTypes.shape({
		_id: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired,
	}),
};

const mapStateToProps = state => ({
	isAuthenticated: state.auth.isAuthenticated,
	user: state.auth.user
});

export default connect(mapStateToProps)(Home);
