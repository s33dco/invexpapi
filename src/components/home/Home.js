import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import About from './About';
import Dashboard from './Dashboard';

const Home = ({ isAuthenticated }) => {
	return (
		<div className="container">
			{isAuthenticated ? <Dashboard /> : <About />}
		</div>
	);
};

Home.propTypes = {
	isAuthenticated: PropTypes.bool.isRequired,
	token: PropTypes.string
};

const mapStateToProps = state => ({
	isAuthenticated: state.auth.isAuthenticated
});

export default connect(mapStateToProps)(Home);
