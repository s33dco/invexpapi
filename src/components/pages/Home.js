import React from 'react';

const Home = () => {
	// isLoggedIn ? <Dashboard /> : <About />
	return <h1>Home</h1>;
};

export default Home;
