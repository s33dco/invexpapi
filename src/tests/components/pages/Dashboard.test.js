import React from 'react';
import { shallow } from 'enzyme';
import Home from '../../../components/pages/Dashboard';

let wrapper;

beforeEach(() => {
	wrapper = shallow(<Home />);
});

describe('Dashboard', () => {
	it('should render Dashboard correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
