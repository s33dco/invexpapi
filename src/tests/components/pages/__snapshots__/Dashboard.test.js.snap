// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render Dashboard correctly 1`] = `ShallowWrapper {}`;
